### PR TITLE
Add agent for Skala-R virtualization platform

### PR DESCRIPTION
**Pull request by vbp1**
This adds an agent for the Skala-R hyper-converged virtualization platform (https://www.skala-r.ru/) via the REST API.

### CODE REVIEW

File: agents/skala-r/fence_skalar.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+import json
+import logging
+
+from requests.exceptions import ConnectionError
+
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+cookie = None
+
+RELEASE_VERSION="1.1"
```
Comment (oalbrigt) [inline agents/skala-r/fence_skalar.py:21]: You can remove these, as we've removed them from the other agents: https://github.com/ClusterLabs/fence-agents/pull/131/files.

Comment (vbp1) [reply agents/skala-r/fence_skalar.py:21]: [root@node01 ~]# fence_skalar -V
Traceback (most recent call last):
  File "/usr/sbin/fence_skalar", line 231, in <module>
    main()
  File "/usr/sbin/fence_skalar", line 202, in main
    show_docs(options, docs)
  File "/usr/share/fence/fencing.py", line 753, in show_docs
    print(__main__.RELEASE_VERSION, __main__.BUILD_DATE)
AttributeError: 'module' object has no attribute 'RELEASE_VERSION'

Comment (vbp1) [reply agents/skala-r/fence_skalar.py:21]: > You can remove these, as we've removed them from the other agents: https://github.com/ClusterLabs/fence-agents/pull/131/files.

[root@node01 ~]# fence_skalar -V
Traceback (most recent call last):
  File "/usr/sbin/fence_skalar", line 231, in <module>
    main()
  File "/usr/sbin/fence_skalar", line 202, in main
    show_docs(options, docs)
  File "/usr/share/fence/fencing.py", line 753, in show_docs
    print(__main__.RELEASE_VERSION, __main__.BUILD_DATE)
AttributeError: 'module' object has no attribute 'RELEASE_VERSION'

Comment (oalbrigt) [reply agents/skala-r/fence_skalar.py:21]: Oh. It seems like you have an old version of fencing.py on your nodes.

Comment (vbp1) [reply agents/skala-r/fence_skalar.py:21]: > Oh. It seems like you have an old version of fencing.py on your nodes.
Yes, already realized it;) Unfortunately this script should works in old env too.
But I'm going to make what you have advised.

Comment (oalbrigt) [reply agents/skala-r/fence_skalar.py]: Thanks.

My mistake. You can also remove the 2 lines below it.

Comment (vbp1) [reply agents/skala-r/fence_skalar.py]: Ready for the review. Removed RELEASE_VERSION + a few improvement related to ssl

Comment (vbp1) [reply agents/skala-r/fence_skalar.py]: > Thanks.
> 
> My mistake. You can also remove the 2 lines below it.

Done.

File: agents/skala-r/fence_skalar.py [hunk 2]
```diff
@@ -206,6 +201,7 @@ def main():
     
     if "--ssl" in options or "--ssl-secure" in options or "--ssl-insecure" in options:
         proto = "https://"
+        ssl_verify = True
     else:
         proto = "http://"
         ssl_verify = False
```
Comment (oalbrigt) [inline agents/skala-r/fence_skalar.py:207]: You probably want this in an `elif "--ssl-insecure" in options:` as it only affects SSL/TLS requests, and you could bump `https://` outside the if/else as I dont think anyone wants to use things without SSL these days.

Comment (vbp1) [reply agents/skala-r/fence_skalar.py:206]: Fixed.

>I dont think anyone wants to use things without SSL these days.
You would be suprised, but yes sometimes someone wants to go without SSL.

**Comment by oalbrigt** [comment]
ok to test

File: agents/skala-r/fence_skalar.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+    proto = "https://"
+    if "--ssl" in options or "--ssl-secure" in options:
+        ssl_verify = True
+    elif "--ssl-insecure" in options:
+        ssl_verify = False
+        urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+    else:
+        proto = "http://"
+        ssl_verify = False
+        urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
```
Comment (oalbrigt) [inline agents/skala-r/fence_skalar.py:211]: You can remove these 2 lines, as they only apply for HTTPS.

Comment (vbp1) [reply agents/skala-r/fence_skalar.py:211]: Generally yes, but if web server redirects to https, requests.Session tries to make a request to https and to this request these params will be useful.

Comment (oalbrigt) [reply agents/skala-r/fence_skalar.py:211]: Ah perfect.

**Comment by oalbrigt** [comment]
Can you rename the directory to `skalar`, so it can also be included in the build-process when selecting agents (e.g. `./configure --with-agents="skalar  mpath scsi"`?

**Comment by vbp1** [comment]
> Can you rename the directory to `skalar`, so it can also be included in the build-process when selecting agents (e.g. `./configure --with-agents="skalar mpath scsi"`?


Sure. Done.

**Comment by oalbrigt** [comment]
LGTM. Thanks.

**Pull request merged**